src/components/StatePicker.js

```diff
@@ -1,67 +1,102 @@
-import _ from 'underscore';
-import React, {forwardRef} from 'react';
+import lodashGet from 'lodash/get';
+import React, {
+    useState, useEffect, useCallback, useMemo,
+} from 'react';
+import {View} from 'react-native';
+import {useRoute} from '@react-navigation/native';
 import PropTypes from 'prop-types';
-import Picker from './Picker';
+import compose from '../libs/compose';
+import withNavigation from './withNavigation';
+import styles from '../styles/styles';
 import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import MenuItemWithTopDescription from './MenuItemWithTopDescription';
+import Navigation from '../libs/Navigation/Navigation';
+import ROUTES from '../ROUTES';
+import FormHelpMessage from './FormHelpMessage';
 
 const propTypes = {
-    /** The label for the field */
-    label: PropTypes.string,
-
-    /** A callback method that is called when the value changes and it receives the selected value as an argument. */
-    onInputChange: PropTypes.func.isRequired,
-
-    /** The value that needs to be selected */
-    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-
-    /** The ID used to uniquely identify the input in a Form */
-    inputID: PropTypes.string,
-
-    /** Saves a draft of the input value when used in a form */
-    shouldSaveDraft: PropTypes.bool,
-
-    /** Callback that is called when the text input is blurred */
-    onBlur: PropTypes.func,
+    /** Current State from user address  */
+    stateISO: PropTypes.string,
 
     /** Error text to display */
     errorText: PropTypes.string,
 
+    /** Default value to display */
+    defaultValue: PropTypes.string,
+
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
-    label: '',
-    value: undefined,
+    stateISO: '',
     errorText: '',
-    shouldSaveDraft: false,
-    inputID: undefined,
-    onBlur: () => {},
+    defaultValue: '',
 };
 
-const StatePicker = forwardRef((props, ref) => {
-    const STATES = _.chain(props.translate('allStates'))
-        .sortBy(state => state.stateName.toLowerCase())
-        .map(state => ({value: state.stateISO, label: state.stateName}))
-        .value();
+function BaseStatePicker(props) {
+    const route = useRoute();
+    const stateISO = props.stateISO;
+    const [stateTitle, setStateTitle] = useState(stateISO);
+    const paramStateISO = lodashGet(route, 'params.stateISO');
+    const navigation = props.navigation;
+    const onInputChange = props.onInputChange;
+    const defaultValue = props.defaultValue;
+    const translate = props.translate;
```

Comment:
Too many unnecessary declarations here, to improve readability please destruct props in component or call functions with just a simple approach `props.navigation.navigate(...)`

Reply:
Yeah, I talked internally about this but seems that we aren't gonna do [destruction](https://expensify.slack.com/archives/C01GTK53T8Q/p1680718704040129), but probably we can bring it back in discussion

Reply:
This topic is mainly about `defaultProps` with destructing, but let's leave destructing until that is clarified.
I propose to use simple syntax `props.navigation.navigate` to reduce variable declaration, which will improve the hook's readability right now, it looks a bit messy